llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9971,8 +9968,9 @@ bool LoopVectorizePass::processLoop(Loop *L) {
       InnerLoopUnroller Unroller(L, PSE, LI, DT, TLI, TTI, AC, ORE, IC, &LVL,
                                  &CM, BFI, PSI, Checks);
 
-      VPlan &BestPlan = LVP.getBestPlan();
-      assert(BestPlan.hasScalarVFOnly() &&
+      ElementCount BestVF = LVP.getBestVF();
+      VPlan &BestPlan = LVP.getBestPlanFor(BestVF);
+      assert(BestVF.isScalar() &&
              "VPlan cost model and legacy cost model disagreed");
       LVP.executePlan(VF.Width, IC, BestPlan, Unroller, DT, false);
```

Comment:
```suggestion
      LVP.executePlan(BestVF, IC, BestPlan, Unroller, DT, false);
```
following the typical pattern of having `LVP.executePlan(VF, IC, Plan, ...)` follow `VPlan &Plan = LVP.getPlanFor(VF)`.